akka-sse/src/main/scala/de/heikoseeberger/akkasse/ServerSentEventParser.scala

```diff
@@ -26,15 +28,24 @@ private object ServerSentEventParser {
 
   private final val Event = "event"
 
+  private final val Id = "id"
+
+  private final val Retry = "retry"
+
   private val linePattern = """([^:]+): ?(.*)""".r
 
   private def parseServerSentEvent(lines: Seq[String]) = {
     val valuesByField = lines
-      .collect { case linePattern(field @ (Data | Event), value) => field -> value }
+      .collect { case linePattern(field @ (Data | Event | Id | Retry), value) => field -> value }
       .groupBy(_._1)
     val data = valuesByField.getOrElse(Data, Vector.empty).map(_._2).mkString(LF)
     val event = valuesByField.getOrElse(Event, Vector.empty).lastOption.map(_._2)
-    ServerSentEvent(data, event)
+    val idField = valuesByField.getOrElse(Id, Vector.empty).lastOption.map(_._2) collect {
+      case "" => ServerSentEvent.ResetId
+      case id => ServerSentEvent.Id(id)
+    }
+    val retry = valuesByField.getOrElse(Retry, Vector.empty).lastOption.map(_._2).map(s => Try(s.trim.toInt).toOption).flatten
```

Comment:
Coding style – see above.
Also only use `Try` as parameter or return type, not for convenient implementation, because it unnecessarily allocates. Instead, use try-catch.